### PR TITLE
UI Team: projectmgr endpoint

### PR DESCRIPTION
**Pull request by daysleeperx**
- added ofbiz projectmgr plugin: https://cwiki.apache.org/confluence/display/OFBIZ/Project+Management
 
- added ofbiz-jersey plugin for REST implementation(maybe an alternative to Camel?): https://github.com/girishvasmatkar/ofbiz-jersey
 
- Resource implemented can be found here:
`plugins/ofbiz-jersey/src/main/java/org/apache/ofbiz/jersey/resource/ProjectResource.java`

- ProjectTaskCmd object: `plugins/projectmgr/src/main/java/org/apache/ofbiz/project/model/ProjectTaskCmd.java`

- initial version of Workspace plugin was added

- added custom java services to projectmgr:
`plugins/projectmgr/src/main/java/org/apache/ofbiz/project/services/ProjectCustomServices.java`

**Please note that all other files came with the projectmgr plugin and were not implemented by our team.**

### CODE REVIEW

File: plugins/workspace/src/main/java/com/taltech/workspace/services/WorkspaceServices.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        try {
+            GenericValue workspace = delegator.makeValue("Workspace");
+            workspace.setNextSeqId();
+            workspace.setNonPKFields(context);
+            workspace = delegator.create(workspace);
+            result.put("workspaceId", workspace.getString("workspaceId"));
+            Debug.log("Workspace record created successfully with workspaceId:" + workspace.getString("workspaceId"));
+        } catch (GenericEntityException e) {
+            Debug.logError(e, module);
+            return ServiceUtil.returnError("Error in creating record in Workspace entity ......." + "." + module);
```
Comment (ghost) [inline plugins/workspace/src/main/java/com/taltech/workspace/services/WorkspaceServices.java]: ......." + "." ?

File: plugins/projectmgr/src/main/java/org/apache/ofbiz/project/Various.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    public static double calculateActualHours(Delegator delegator, String timesheetId) {
+        double actualHours = 0.00;
+        if (timesheetId != null) {
+            try {
+                List<GenericValue> actuals = EntityQuery.use(delegator).from("TimeEntry").where("timesheetId", timesheetId).queryList();
+                if (actuals.size() > 0) {
+                    for (GenericValue actual : actuals) {
+                        Double hour = (Double) actual.get("hours");
+                        double hours = hour;
```
Comment (ghost) [inline plugins/projectmgr/src/main/java/org/apache/ofbiz/project/Various.java]: Rather weird Double handling in this file.

File: plugins/workspace/src/main/java/com/taltech/workspace/services/WorkspaceServices.java (hunk 1 above)
Comment (daysleeperx) [reply plugins/workspace/src/main/java/com/taltech/workspace/services/WorkspaceServices.java]: fixed

File: plugins/projectmgr/src/main/java/org/apache/ofbiz/project/Various.java (hunk 2 above)
Comment (daysleeperx) [reply plugins/projectmgr/src/main/java/org/apache/ofbiz/project/Various.java]: agreed. java should unbox on assignment. fixed.

**Pull request merged**